src/DataProviders/SqlDataProvider/SqlDataProvider.ts

```diff
@@ -28,12 +31,37 @@ export class SqlDataProvider implements IDataProvider {
         };
     }
 
+    private static convertServiceDbItemToServiceInfo(serviceDbItem: HazzatDbSchema.IService): IServiceInfo {
+        const contentCount: StringMap<number> = {};
+        contentCount[FormatType.Text] = serviceDbItem.Text_Count;
+        contentCount[FormatType.Hazzat] = serviceDbItem.Hazzat_Count;
+        contentCount[FormatType.VerticalHazzat] = serviceDbItem.VerticalHazzat_Count;
+        contentCount[FormatType.Music] = serviceDbItem.Music_Count;
+        contentCount[FormatType.Audio] = serviceDbItem.Audio_Count;
+        contentCount[FormatType.Video] = serviceDbItem.Video_Count;
+        contentCount[FormatType.Information] = serviceDbItem.Information_Count;
+
+        return {
+            contentCount,
+            id: serviceDbItem.ItemId,
+            name: serviceDbItem.Structure_Name,
+            order: serviceDbItem.Service_Order
+        };

```

Comment:
Perhaps a dedicated mapper class would be more appropriate here?

Reply:
Not sure what you mean.  Example?

Reply:
Another class such as GetServicesRepository.ts to call the SqlDataProvider.cs and provide a repository layer/mapping abstraction.

Reply:
Not sure I follow.  The stored procedure already returns the counts for all the different formats

Reply:
Maybe I'm the one who's not following 😁 LGTM      